CodeConverter/CSharp/TypeConversionAnalyzer.cs

```diff
@@ -273,15 +289,20 @@ private ITypeSymbol GetCSType(ITypeSymbol vbType, VBSyntax.ExpressionSyntax vbNo
                 return true;
             }
             if (isConvertToString || vbConversion.IsNarrowing) {
-                typeConversionKind = vbConvertedType.IsEnumType() && !csConversion.Exists ? TypeConversionKind.EnumConversionThenCast : TypeConversionKind.Conversion;
+                typeConversionKind = nullableVbConvertedType.IsEnumType() && !csConversion.Exists
+                    ? TypeConversionKind.EnumConversionThenCast
+                    : TypeConversionKind.Conversion;
                 return true;
             }
-        } else if (csConversion.IsExplicit && csConversion.IsEnumeration || csConversion.IsBoxing) {
-            typeConversionKind = TypeConversionKind.NonDestructiveCast;
+        } else if (vbConversion.IsNarrowing && vbConversion.IsNullableValueType && isConvertFractionalToInt) {
```

Comment:
This case was just moved down because you can actually explicitly cast fractional number to enum in C#:
```csharp
var x = (TestEnum)1.8f;
```

but it will not produce the same result. VB rounds the value before casting so we can treat it as ```FractionalNumberRoundThenCast```.